src/lib/Form/Factory/FormFactory.php

```diff
@@ -699,6 +701,21 @@ public function createRole(
         return $this->formFactory->createNamed($name, RoleCreateType::class, $data);
     }
 
+    /**
+     * @param RoleCopyData $data
+     * @param string|null $name
+     *
+     * @return FormInterface
+     */
+    public function copyRole(
+        RoleCopyData $data,
+        ?string $name = null
+    ): FormInterface {
+        $name = $name ?: sprintf('copy-role-%d', $data->getClonedRole()->id);
+
+        return $this->formFactory->createNamed($name, RoleCopyType::class, $data);
+    }
```

Comment:
Didn't we decide to stop evolving this form factory as it was not SOLID? Or is it still ok?
Question to: @Nattfarinn @adamwojs @ViniTou @mikadamczyk  @webhdx as you were involved in a discussion, I think.

Reply:
You are right @alongosz. `Symfony\Component\Form\FormFactoryInterface` should be used directly if needed.

Reply:
Yeah it's not recommended to add any new methods to the `FormFactory` unless you want to display multiple forms on the same page and you need unique naming.